From 355ad33087c1c683458b60a6bc9e1f89623ee275 Mon Sep 17 00:00:00 2001
From: Natanael Copa <natanael.copa@docker.com>
Date: Thu, 14 Apr 2016 17:29:12 +0200
Subject: [PATCH] Fix detection of git commit during build from tarball

Distro packagers will often use the tarball to build a package and have
the build script for the package in git. To avoid that the docker build
script picks up the git commit from the distro repo we also check for a
directory named .git before check for -unsupported builds.

Signed-off-by: Natanael Copa <natanael.copa@docker.com>
---
 hack/make.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hack/make.sh b/hack/make.sh
index 58561e2..41a53f6 100755
--- a/hack/make.sh
+++ b/hack/make.sh
@@ -79,7 +79,7 @@ DEFAULT_BUNDLES=(
 )
 
 VERSION=$(< ./VERSION)
-if command -v git &> /dev/null && git rev-parse &> /dev/null; then
+if command -v git &> /dev/null && [ -d .git ] && git rev-parse &> /dev/null; then
 	GITCOMMIT=$(git rev-parse --short HEAD)
 	if [ -n "$(git status --porcelain --untracked-files=no)" ]; then
 		GITCOMMIT="$GITCOMMIT-unsupported"
